packages/@aws-cdk/aws-rds/lib/cluster.ts

```diff
@@ -595,6 +596,13 @@ export class DatabaseCluster extends DatabaseClusterBase {
         // Instance properties
         dbInstanceClass: databaseInstanceType(instanceType),
         publiclyAccessible,
+        enablePerformanceInsights: instanceProps.enablePerformanceInsights,
+        performanceInsightsKmsKeyId: instanceProps.enablePerformanceInsights
+          ? instanceProps.performanceInsightEncryptionKey?.keyArn
```

Comment:
I wonder - should passing a `performanceInsightEncryptionKey` imply `enablePerformanceInsights` to `true`? Or, alternatively, should we validate that if `performanceInsightEncryptionKey` was passed, then `enablePerformanceInsights` is `true`? Just some food for thought.

Reply:
Yeah, I definitely had that thought too. I ended up opting for "keep existing behavior", but very much willing to improve things instead. I think auto-enabling performance insights if either the key or retention period are specified makes sense, and erroring if those fields are provided with performance insights explicitly disabled. I'll update to that effect.